libsolidity/codegen/ir/IRLValue.cpp

```diff
@@ -61,59 +61,46 @@ IRStorageItem::IRStorageItem(
 	unsigned offset;
 	std::tie(slot, offset) = _context.storageLocationOfVariable(_varDecl);
 	m_slot = toCompactHexWithPrefix(slot);
-	m_offset = offset;
+	m_offset = to_string(offset);
+	solAssert(!m_offset.empty(), "");
+	solAssert(!m_slot.empty(), "");
 }
 
 IRStorageItem::IRStorageItem(
 	IRGenerationContext& _context,
 	string _slot,
-	unsigned _offset,
+	string _offset,
 	Type const& _type
 ):
 	IRLValue(_context, &_type),
 	m_slot(move(_slot)),
-	m_offset(_offset)
+	m_offset(move(_offset))
 {
+	solAssert(!m_offset.empty(), "");
+	solAssert(!m_slot.empty(), "");
 }
 
 string IRStorageItem::retrieveValue() const
 {
 	if (!m_type->isValueType())
 		return m_slot;
 	solUnimplementedAssert(m_type->category() != Type::Category::Function, "");
-	return m_context.utils().readFromStorage(*m_type, m_offset, false) + "(" + m_slot + ")";
+	return m_context.utils().dynamicReadFromStorage(*m_type, false) + "(" + m_slot + ", " + m_offset + ")";
```

Comment:
You're effectively making *all* ``IRStorageItem``s dynamic, aren't you? So you add the overhead of a dynamic read, even if the offset is known in advance? One could trust in the optimizer to be able to optimize away the constant function argument... currently it can't do that... So I think it might be better for now, if ``IRStorageItem`` still supports static offsets and e.g. uses ``readFromStorage`` for static offsets and ``dynamicReadFromStorage```for dynamic offsets. What do you think?

Reply:
Yeah, that's true. I'll change it to handle both